lib/internal/test_runner/test.js

```diff
@@ -29,12 +31,27 @@ const kMultipleCallbackInvocations = 'multipleCallbackInvocations';
 const kParentAlreadyFinished = 'parentAlreadyFinished';
 const kSubtestsFailed = 'subtestsFailed';
 const kTestCodeFailure = 'testCodeFailure';
+const kTestTimeoutFailure = 'testTimeoutFailure';
 const kDefaultIndent = '    ';
 const noop = FunctionPrototype;
 const isTestRunner = getOptionValue('--test');
 const testOnlyFlag = !isTestRunner && getOptionValue('--test-only');
 // TODO(cjihrig): Use uv_available_parallelism() once it lands.
 const rootConcurrency = isTestRunner ? cpus().length : 1;
+const rootTimeout = 30_0000;
```

Comment:
IMHO timeouts are not a good mechanism for performance testing, and the only reason to have (default) timeouts is to detect when a test is truly stuck and will never succeed. That is a somewhat rare scenario though, so a large timeout rarely negatively impacts developer experience. On the other hand, timeouts can very negatively impact developer experience when tests pass on fast machines but fail on slower ones. If we really want a default timeout, I'd suggest making it a full minute or more.

Reply:
I think in bdd or similar approaches it is not a rare scenario at all, but I would love hearing more opinions about this